Ignore complaints about dead code

greg0ire opened the pull request:
~In the code of these constructors, we are checking the type of the 'value' property of the '$values' array.
Let us reflect the fact that we are not assuming the type in the phpdoc. I think this is OK because since both of these classes are used as Doctrine annotations, there is no way for such precise types to be useful for the end user anyway. I chose that over creating a baseline for a project that does not have one yet.~



We've declared types in phpdoc, but they are not enforced by type
declarations, so we are enforcing them with checks. Psalm complains it
is dead code because it assumes the types can be trusted.

This addresses a Psalm issue I spotted while reviewing #494

derrabus commented:
What issue does this fix exactly?

greg0ire commented:
The issue is not very clearly expressed:

```
ERROR: NoValue - lib/Doctrine/Common/Annotations/Annotation/Enum.php:51:49 - All possible types for this argument were invalidated - This may be dead code (see https://psalm.dev/179)
                    is_object($var) ? get_class($var) : gettype($var)


ERROR: NoValue - lib/Doctrine/Common/Annotations/Annotation/Target.php:77:61 - All possible types for this argument were invalidated - This may be dead code (see https://psalm.dev/179)
                    is_object($values['value']) ? get_class($values['value']) : gettype($values['value'])
```

Since the phpdoc says `scalar`, and we are inside a condition on `!is_scalar(…)`, the inferred type is `never`, and that should never happen, so Psalm suspects there is dead code.

SenseException commented:
While there isn't a baseline file, you can find `ignoreErrors` directly in [phpstan.neon](https://github.com/doctrine/annotations/blob/2.0.x/phpstan.neon).

derrabus commented:
But those "errors" happen during input validation where we handle exactly the cases that we declare unacceptable in the doc blocks that you're about to change. Psalm complaining about input validation is just Psalm being Psalm.

I would simply ignore them.

greg0ire commented:
Ok, let me adjust.

greg0ire committed:
Ignore complaints about dead code

We've declared types in phpdoc, but they are not enforced by type
declarations, so we are enforcing them with checks. Psalm complains it
is dead code because it assumes the types can be trusted.

greg0ire commented:
@SenseException I did the Psalm equivalent of what you mentioned.

derrabus reviewed (approved):
LGTM. The PR title needs to be adjusted though.
